x/marketmap/types/msg.go

```diff
@@ -11,8 +11,30 @@ var (
 	_ sdk.Msg = &MsgUpdateMarkets{}
 	_ sdk.Msg = &MsgParams{}
 	_ sdk.Msg = &MsgRemoveMarketAuthorities{}
+	_ sdk.Msg = &MsgUpsertMarkets{}
 )
 
+// ValidateBasic asserts that the authority address in the upsert-markets message is formatted correctly.
+// If also verifies that all markets w/in the message are valid, if no markets are present it returns an error.
+func (m *MsgUpsertMarkets) ValidateBasic() error {
+	// validate signer address
+	if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
+		return err
+	}
+
+	if len(m.Markets) == 0 {
+		return fmt.Errorf("no markets to upsert")
+	}
+
+	for _, market := range m.Markets {
+		if err := market.ValidateBasic(); err != nil {
+			return err
+		}
+	}
```

Comment:
Could we check for duplicate markets here?

Reply:
We aren't doing this for the other msg-types (`Create` / `Update`), but I can add